lib/modules/manager/gomod/artifacts.ts

```diff
@@ -134,9 +134,9 @@ export async function updateArtifacts({
   const vendorDir = upath.join(goModDir, 'vendor/');
   const vendorModulesFileName = upath.join(vendorDir, 'modules.txt');
   const useVendor =
-    !config.postUpdateOptions?.includes('gomodSkipVendor') &&
-    (await readLocalFile(vendorModulesFileName)) !== null;
-
+    (!config.postUpdateOptions?.includes('gomodSkipVendor') &&
+      (await readLocalFile(vendorModulesFileName)) !== null) ||
+    config.postUpdateOptions?.includes('gomodVendor');
```

Comment:
```suggestion
    config.postUpdateOptions?.includes('gomodVendor') ||
    ((await readLocalFile(vendorModulesFileName)) !== null &&
      !config.postUpdateOptions?.includes('gomodSkipVendor'));
```

I think this ordering is easier to understand/maintain. I'm not sure if prettier will want the outer () there or not.

Reply:
That was my initial approach but the linter shut me down for using OR on a nullish value

Reply:
How about now? Changed || to ??

Reply:
For some reason the ?? operator failed tests but will try to figure it out later today

Reply:
ok figured the issue, now that the condition is before just needed to remove the mock that reads the vendor file

Reply:
will add ignoreConditionalTests to allow `??` in conditions